scripts/sanitycheck

```diff
@@ -1687,11 +1687,11 @@ class TestCase:
 
     def get_unique(self, testcase_root, workdir, name):
 
-        if ZEPHYR_BASE in testcase_root:
+        real_zephyr_base = os.path.realpath(ZEPHYR_BASE)
```

Comment:
Why not normalize the path when ZEPHYR_BASE is assigned, once and for all?

Reply:
Good question. While not larger in lines of code, this would have massively larger side effects which may or may not be desired and/or buggy.
ZEPHYR_BASE is referred to ~ 20 times in this script but no other seems to require normalization, this small fix is enough to make everything work for me.

Reply:
There's also a consistency issue. sanitycheck is essentially a wrapper on top of the main build system and in number of places it uses ZEPHYR_BASE to generate path arguments to other tools of the build. Let's pretend my `/home/march-db` is a symbolic link to `/nfs/serverA/home/marc-hb`. I would really not like sanitycheck to invoke the build with `/nfs/serverA/home/marc-hb` while I invoke it with `/home/march-db`, that would be at least a source of confusion and deception and possibly worse.

Reply:
Hm, I don't think I understand the change well enough to review then. I'll drop my -1 and see what @nashif says :)